src/engine/Union.cpp

```diff
@@ -181,10 +188,22 @@ void Union::computeUnion(
       // it would not be possible to call the function due to not matching
       // columns.
       if constexpr (LEFT_WIDTH == OUT_WIDTH) {
-        res.insert(res.end(), left.begin(), left.end());
+        size_t numChunks = left.size() / chunkSize;
+        for (size_t i = 0; i < numChunks; ++i) {
+          res.insert(res.end(), left.begin() + i * chunkSize,
+                     left.begin() + (i + 1) * chunkSize);
+          checkTimeout();
+        }
+        res.insert(res.end(), left.begin() + numChunks * chunkSize, left.end());
       }
     } else {
+      size_t numHandled = 0;
       for (const auto& l : left) {
+        numHandled += 1;
```

Comment:
Factor this really common pattern out into a helper lambda etc. that belongs to the Timer interface.